### DISCUSSION TITLE
「开源之史」系列之十/四：浏览器竞争下的创新模式——Mozilla公共许可 - 「开源之道」

### DISCUSSION THREAD

**Discussion by giscus** [OCselected/ocselected.github.io#54] [Announcements]
# 「开源之史」系列之十/四：浏览器竞争下的创新模式——Mozilla公共许可 - 「开源之道」

https://opensourceway.community/posts/history-of-open-source/02-07-01-brower-war-failed-netscape-mpl/

**Comment by ZoomQuiet** [comment]
#是也乎,(￣▽￣)
叕一个刚刚开头的故事, 
参考: [浏览器战争](http://s5.zoomquiet.top/131001-war-of-browser/index.html)

当年在折腾猎豹浏览器时, 专门回顾了一下这个故事;

和其它开源项目不同, Mozilla 是完全被逼的,
试想, 市场份额在不到两年时间里从 将近 100% 变成不到一成,
无论谁都感觉药丸...

当然, 危机危机, 危险中永远包含机遇,
这次, 是被自由软件叛徒: Raymond 戳中了:

- 反正已经死了
- 不如鲸落一下?

也实在是 IE 当年不够打, 自身开发团队太微软,
在 Netscape -> Mozilla -> FireFox 过程中,
根本来不及反应, PR 团队不作为,
结果, 光是 Mozilla 社区中艺术家们的宣传画,
就已经将整个儿互联网给感染了...

![](https://ipic.zoomquiet.top/2022-06-07-ff-eat-ie.gif)

类似非常有倾向性的图标/图片/口号/... 不知道有多少...
而且有专门的 [Affiliates homepage - Spread Firefox](http://smm.zoomquiet.top/data/20050106180021/index.html)
推广站, 整个儿社区发动空前的资源在所有层面针对 IE 进行战斗...

也幸好 FireFox 自身给力, 
积极迎合使用者的述求, 并创造性的发明了浏览器扩展这一 UGC 空间, 
其实, 也就是浏览器软件次级插件开源空间;

瞬间互联网中到处都是:

- 为什么使用火狐浏览器的17个理由
- XX年最值得安装的50个火狐插件
- ...

最高峰时, 俺的 FF 安装了超过100个插件,
整个浏览器已经完全看不出是什么品种了...

是的, Mozilla 的创新不在法律层面, 
而是首次有商业公司, 先放弃了商业利益, 才又被市场狠狠塞回了市场份额;
可惜一直没找到合理的变现渠道,
直到 FF 再次衰败, 最大的收入依然是 内置google搜索 的流量广告...

所以, 这故事, 真的要讲下去,
是要对比 Chrome 成功故事的...

那是另外一个霜之哀伤了...
